stdlib/public/core/Runtime.swift.gyb

```diff
@@ -121,8 +121,9 @@ func _stdlib_atomicCompareExchangeStrongInt${bits}(
   expected: UnsafeMutablePointer<Int${bits}>,
   desired: Int${bits}) -> Bool {
   return _stdlib_atomicCompareExchangeStrongUInt${bits}(
-    object: UnsafeMutablePointer(target),
-    expected: UnsafeMutablePointer(expected),
+    object: unsafeBitCast(target, to: UnsafeMutablePointer<UInt${bits}>.self),
+    expected: unsafeBitCast(expected,
+      to: UnsafeMutablePointer<UInt${bits}>.self),
```

Comment:
Given the number of unsafe bit casts required, I think a better approach is just to call the builtin directly.